### PR TITLE
Issue #1362

### LINKED ISSUE #1362: Crash when getting URI MIME type.
#### Software and hardware versions 
Collect v1.9.1

#### Problem description
Crash in `FormEntryActivity` when getting intent URI MIME type.

#### Steps to reproduce the problem
Unknown, from Play Store [here](https://play.google.com/apps/publish/?dev_acc=00659255839475882647#AndroidMetricsErrorsPlace:p=org.odk.collect.android&appVersion=PRODUCTION&clusterName=apps/org.odk.collect.android/clusters/9956b8d6) (requires credentials). It only affects Android 6.0 but on a range of devices.

The line that crashes is `String uriMimeType = getContentResolver().getType(uri);`. Either the content resolver can be `null` or `getType` dereferences `uri`?

#### Expected behavior
No crash.

#### Other information 
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2572)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2654)
  at android.app.ActivityThread.-wrap11 (ActivityThread.java)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1488)
  at android.os.Handler.dispatchMessage (Handler.java:111)
  at android.os.Looper.loop (Looper.java:207)
  at android.app.ActivityThread.main (ActivityThread.java:5728)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:789)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:679)
Caused by: java.lang.NullPointerException: 
  at com.android.internal.util.Preconditions.checkNotNull (Preconditions.java:60)
  at android.content.ContentResolver.getType (ContentResolver.java:336)
  at org.odk.collect.android.activities.FormEntryActivity.onCreate (FormEntryActivity.java:367)
  at android.app.Activity.performCreate (Activity.java:6360)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1113)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2519)
```

### PR DESCRIPTION
**Pull request by akshay-ap**
Closes #1362

### CODE REVIEW

**Comment by grzesiek2010** [comment]
@akshay-ap 
it's your first pr so Congrats of course :clap::clap::clap:

The first thing you should know we use static code analysis to keep the code consistent. It's [checkstyle](http://checkstyle.sourceforge.net/), [lint](http://tools.android.com/tips/lint-checks) and recently we also added pmd and findbusg. But now checkstyle and maybe lint are important for you (That's why your branch failed on CircleCI).

You can click on `details` above to find more information
https://circleci.com/gh/opendatakit/collect/814?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

It's a part of your logs
> Building 0% > :collect_app:checkstyle[ant:checkstyle] [ERROR] /home/ubuntu/collect/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java:368:37: WhitespaceAround: '=' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
> Building 0% > :collect_app:checkstyle[ant:checkstyle] [ERROR] /home/ubuntu/collect/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java:371:19: WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[ant:checkstyle] [ERROR] /home/ubuntu/collect/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java:371:23: WhitespaceAround: '!=' is not preceded with whitespace. [WhitespaceAround]
[ant:checkstyle] [ERROR] /home/ubuntu/collect/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java:371:25: WhitespaceAround: '!=' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[ant:checkstyle] [ERROR] /home/ubuntu/collect/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java:373:17: '}' at column 17 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally). [RightCurly]
[ant:checkstyle] [ERROR] /home/ubuntu/collect/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java:375:17: '{' at column 17 should be on the previous line. [LeftCurly]

So the first thing you should do is fixing all those errors.
Don't hesitate to ask if you don't understand anything.

**Comment by lognaturel** [comment]
As @grzesiek2010 said, thanks for your contribution, @akshay-ap!

In this case, the user doesn't really have to be notified of anything so it's ok to just have the `if` branch you had and not the `else` branch. Additionally, since the code is nice and clear, no comments are needed.

**Comment by akshay-ap** [comment]
Ok. I am working on it. Do I need to file another Pull request?

**Comment by grzesiek2010** [comment]
@akshay-ap No, you don't need. You can edit this one by adding new commits or you can also edit the existed commit and use `git push` command with `--force` option.

**Comment by akshay-ap** [comment]
Thanks for suggestions. Shall I commit again and issue a Pull request?

File: collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java [hunk 1]
```diff
@@ -365,7 +365,11 @@ public void onClick(View v) {
             Intent intent = getIntent();
             if (intent != null) {
                 Uri uri = intent.getData();
-                String uriMimeType = getContentResolver().getType(uri);
+                String uriMimeType = null;
+
+                if(uri != null) {
```
Comment (shobhitagarwal1612) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java]: Please add a `space` ' ' between `if` and `(`

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
Thank you @akshay-ap and congrats on your first merged pull request!! 👏🏆

**Pull request merged**